extension/devtools/build/flutter.js

```diff
@@ -0,0 +1,376 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
```

Comment:
There are lots of things in that `build` directory. Do we need them all? Can't we gitignore them and only keep what we need? :)

Reply:
Oh I see, it's a build output of the extension.

Hum... I don't quite like seeing this here, and don't think other packages would either. It sounds like something that begs to be outdated, and pollutes the dev environment a bit.
How was this generated? `flutter build <path_to_extension>` followed by moving the generated build dir there? 

I'd probably ignore it and use a custom tool for publishing which would pre-compile the extension along with publishing

Reply:
Do the instructions I added at https://github.com/flutter/devtools/pull/6402 address your concerns?

Reply:
It explains briefly the problem yes.
Do you think it would be feasible to add a "pub publish --dry-run" validation step for this?

As in fail the validation if a devtool extension is detected but not built

Reply:
@jonasfj would this be possible? Perhaps this would be a sort of "pub extension" where you can add publish validation checks for a package. It seems like this sort of thing would require a pubspec change like we discussed early on. Otherwise I'm not sure where pub would know where to get the additional pub publish validations. 
```
  extensions:
    - devtools
```

Another idea is a more general field for adding custom publish validations
```
publish_checks:
  - devtools_extensions: ^1.0.0
    - verify_extension_build   # maybe this is an executable provided by package `devtools_extensions`?
```

Reply:
Today, `pub publish --dry-run` validations are implemented in `pub`.

I would not oppose to hardcode in a check for `extensions/devtools/..`.

But it would be a one-off hack.

It would of course be awesome, if `pub` would find additional validations from `extension/pub/config.yaml` in dependencies of the package being published.

But there'd be security implications. Maybe we could prompt users for consent...

@sigurdm thoughts?

Reply:
I'm fine with hardcoding it too. I think that makes sense.

Extensibility for that is much less important IMO, and probably in the scope of the "plugins for Dart/Flutter CLI" instead of the devtool.

Reply:
A one-off hardcoded check for DevTools sgtm. We can always remove later and replace with a more extensible validation solution if we create one.

Reply:
Discussing this offline, we're sort of wondering if maybe the number of users who will be publishing devtools extensions is so limited that maybe we shouldn't invest into having special checks for pre-publishing.

Reply:
If we expect the extensions ecosystem to grow, which is the goal, I still think it could be good to have a check. It doesn't have to be anything too complicated, but something simple that checks:
```
if the `extension/devtools/` directory exists
   verify the directory contains a file `config.yaml` and a non-empty `build/` directory
```
if would still be the responsibility of the extension author to ensure they are updating the contents of `build/` as necessary.

Reply:
Filed https://github.com/dart-lang/pub/issues/4026 to track this feature request.